### PR TITLE
Tibor

### PR DESCRIPTION
**Pull request by Tibor22**

### CODE REVIEW

File: src/student.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default class Student {
+  constructor(name, id, githubUsername, email) {
+    ;[this.firstName, this.lastName] = name.split(' ')
```
Comment (sigristarisa) [inline src/student.js:3]: Interesting way to create a full name! Does the semi-colon at the beginning do anything?

File: src/manager.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+export default class Manager {
+  constructor() {
+    this.cohorts = []
+    this.cohortId = 0
+    this.studentId = 0
+  }
+
+  addCohort(name) {
+    if (name.length === 0) return 'Cohort needs to have name'
+    if (this.cohorts.some((cohort) => cohort.name === name))
```
Comment (sigristarisa) [inline src/manager.js:12]: That is a smart way to figure out if there are overlapping cohort names! I'm definitely going to implement that to my code

Comment (PBara7a) [reply src/manager.js:12]: I like this as well, so simple and in one line.

File: src/manager.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          ) {
+            everyMatch.push(student)
+          }
+        }
+      })
+    })
+    const unique = Array.from(new Set(everyMatch))
+    return unique
+  }
+}
```
Comment (Alex90Jennings) [inline src/manager.js:100]: is it possible to refactor this down from 26 lines? Or even split this into more methods that call onto each other, for better readability?

File: src/manager.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  addStudent(cohortName, studentName, github, email) {
+    const currCohort = this.searchCohort(cohortName)
+    if (!currCohort) return 'Cohort Not Found'
+    if (this.studentChecker(studentName, github, email))
+      return 'Student already exists'
+    this.studentId++
+    return currCohort.addStudent(studentName, this.studentId, github, email)
+  }
```
Comment (Alex90Jennings) [inline src/manager.js:39]: you have an addStudent method in both the Cohort and Manager class, can you delete one?

File: src/manager.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+  searchStudent(studentId) {
+    let student
+    this.cohorts.forEach((cohort) => {
+      if (typeof student !== 'object') {
+        student = cohort.students.find((student) => student.id === studentId)
+      }
+    })
+    return student || 'Student not found'
```
Comment (Alex90Jennings) [inline src/manager.js:29]: nice concise code to return either the student or the error message

File: src/manager.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      cohort.students.forEach((student) => {
+        for (const value of Object.values(student)) {
+          if (typeof value !== 'string') break
+
+          if (
+            localeIncludes(value, firstName, {
+              usage: 'search',
+              sensitivity: 'base'
+            }) ||
+            localeIncludes(value, lastName, {
```
Comment (Alex90Jennings) [inline src/manager.js:87]: I've never seen localeIncludes before, I will look it up

**Pull request closed without merging**